internal/util/subjectsetbytype_test.go

```diff
@@ -0,0 +1,69 @@
+package util
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+	"github.com/authzed/spicedb/pkg/tuple"
+)
+
+func RR(namespaceName string, relationName string) *core.RelationReference {
+	return &core.RelationReference{
+		Namespace: namespaceName,
+		Relation:  relationName,
+	}
+}
+
+func TestSubjectByTypeSet(t *testing.T) {
+	assertHasObjectIds := func(s *SubjectByTypeSet, rr *core.RelationReference, expected []string) {
+		wasFound := false
+		s.ForEachType(func(foundRR *core.RelationReference, subjects SubjectSet) {
+			objectIds := make([]string, 0, len(subjects.AsSlice()))
+			for _, subject := range subjects.AsSlice() {
+				require.Nil(t, subject.GetCaveatExpression())
+				require.Empty(t, subject.GetExcludedSubjects())
```

Comment:
let's add caveats to the subjects so we test SubjectByTypeSet handles caveated subjects